packages/plugin-vue/src/handleHotUpdate.ts

```diff
@@ -66,7 +66,7 @@ export async function handleHotUpdate(
     // binding metadata. However, when reloading the template alone the binding
     // metadata will not be available since the script part isn't loaded.
     // in this case, reuse the compiled script from previous descriptor.
-    if (mainModule && !affectedModules.has(mainModule)) {
+    if (mainModule && !scriptChanged) {
```

Comment:
For reviewers: In the past, this part was like:
```js
  const scriptChanged = hasScriptChanged(prevDescriptor, descriptor)
  if (scriptChanged) {
    affectedModules.add(mainModule)
  }
  if (!isEqualBlock(descriptor.template, prevDescriptor.template)) {
    if (!scriptChanged) {
```
https://github.com/vitejs/vite/commit/7883fb2eaae9086a00978d23ec2c36aa80539d3e#diff-9fd57831c2dc5be7b6feb11f8f80bfeca59da8226fac808425dd5460fce7e380R51-R65